File: app/src/main/java/ru/radiationx/anilibria/data/api/release/Release.java
package ru.radiationx.anilibria.data.api.release;

import android.text.Html;

import org.json.JSONArray;
import org.json.JSONObject;

import io.reactivex.Observable;
import ru.radiationx.anilibria.data.Client;

/**
 * Created by radiationx on 05.11.17.
 */

public class Release {

    public Observable<FullRelease> getRelease(int id) {
        return Observable.fromCallable(() -> {
            String url = "http://www.anilibria.tv/api/api.php?action=release&ELEMENT_ID=" + id;
            String response = Client.get().get(url);
            return parseRelease(response);
        });
    }

    private FullRelease parseRelease(String responseText) throws Exception {
        FullRelease release = new FullRelease();

        final JSONObject responseJson = new JSONObject(responseText);
        //item.setId(responseJson.getInt("id"));

        String[] titles = responseJson.getString("title").split(" / ");
        if (titles.length > 0) {
            release.setOriginalTitle(Html.fromHtml(titles[0]).toString());
            if (titles.length > 1) {
                release.setTitle(Html.fromHtml(titles[1]).toString());
            }
        }

        release.setTorrentLink(responseJson.getString("torrent_link"));
        //item.setLink(responseJson.getString("link"));
        release.setImage(responseJson.getString("image"));
        //release.setEpisodes(responseJson.getString("episode"));
        release.setDescription(Html.fromHtml(responseJson.getString("description")).toString());

        JSONArray jsonSeasons = responseJson.getJSONArray("season");
        for (int j = 0; j < jsonSeasons.length(); j++) {
            release.addSeason(jsonSeasons.getString(j));
        }

        JSONArray jsonVoices = responseJson.getJSONArray("voices");
        for (int j = 0; j < jsonVoices.length(); j++) {
            release.addVoice(jsonVoices.getString(j));
        }

        JSONArray jsonGenres = responseJson.getJSONArray("genres");
        for (int j = 0; j < jsonGenres.length(); j++) {
            release.addGenre(jsonGenres.getString(j));
        }

        JSONArray jsonTypes = responseJson.getJSONArray("types");
        for (int j = 0; j < jsonTypes.length(); j++) {
            release.addType(jsonTypes.getString(j));
        }

        return release;
    }
}

File: app/src/main/java/ru/radiationx/anilibria/data/api/release/Release.kt
package ru.radiationx.anilibria.data.api.release;

import android.text.Html
import io.reactivex.Observable
import org.json.JSONObject
import ru.radiationx.anilibria.data.Client
import ru.radiationx.anilibria.data.api.Api
import ru.radiationx.anilibria.data.api.releases.ReleaseItem

/* Created by radiationx on 05.11.17. */

class Release {

    fun getRelease(id: Int): Observable<ReleaseItem> {
        return Observable.fromCallable {
            val url = "http://www.anilibria.tv/api/api.php?action=release&ELEMENT_ID=" + id
            val response = Client.instance.get(url)
            parseRelease(response)
        }
    }

    @Throws(Exception::class)
    private fun parseRelease(responseText: String?): ReleaseItem {
        val release = ReleaseItem()

        val responseJson = JSONObject(responseText)
        //item.setId(responseJson.getInt("id"));

        val titles = responseJson.getString("title").split(" / ".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
        if (titles.size > 0) {
            release.originalTitle = Html.fromHtml(titles[0]).toString()
            if (titles.size > 1) {
                release.title = Html.fromHtml(titles[1]).toString()
            }
        }

        release.torrentLink = Api.BASE_URL + responseJson.getString("torrent_link")
        //item.setLink(responseJson.getString("link"));
        release.image = Api.BASE_URL + responseJson.getString("image")
        //release.setEpisodesCount(responseJson.getString("episode"));
        release.description = Html.fromHtml(responseJson.getString("description")).toString()

        val jsonSeasons = responseJson.getJSONArray("season")
        for (j in 0 until jsonSeasons.length()) {
            release.seasons.add(jsonSeasons.getString(j))
        }

        val jsonVoices = responseJson.getJSONArray("voices")
        for (j in 0 until jsonVoices.length()) {
            release.voices.add(jsonVoices.getString(j))
        }

        val jsonGenres = responseJson.getJSONArray("genres")
        for (j in 0 until jsonGenres.length()) {
            release.genres.add(jsonGenres.getString(j))
        }

        val jsonTypes = responseJson.getJSONArray("types")
        for (j in 0 until jsonTypes.length()) {
            release.types.add(jsonTypes.getString(j))
        }

        val jsonEpisodes = responseJson.getJSONArray("Uppod")
        for (j in 0 until jsonEpisodes.length()) {
            val jsonEpisode = jsonEpisodes.getJSONObject(j)
            val episode = ReleaseItem.Episode()
            episode.title = jsonEpisode.getString("comment")
            episode.urlSd = jsonEpisode.getString("file")
            episode.urlHd = jsonEpisode.getString("filehd")
            release.episodes.add(episode)
        }

        return release
    }
}
